drivers/include/sps30.h

```diff
@@ -0,0 +1,287 @@
+ * |---------|-------------------|------------|-------------------------------|
+ * | Pin 1   | VDD               | 5 V        | should be within +-10 %       |
+ * | Pin 2   | I2C_SDA / UART_RX | SDA*       | config by SPS30_PARAM_I2C_DEV |
+ * | Pin 3   | I2C_SCL / UART_TR | SCL*       | config by SPS30_PARAM_I2C_DEV |
+ * | Pin 4   | SEL               | GND        |                               |
+ * | Pin 5   | GND               | GND        |                               |
+ *
+ *   *The SCL and SDA pins of the SPS30 sensor are open drain so they must be
+ *    pulled high. Consider that internal pull resistors might be too weak.
+ *    So using external 10 kOhm resistors is recommended for that.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Sensirion SPS30 Particulate Matter Sensor
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SPS30_H
+#define SPS30_H
+
+#include <stdbool.h>
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief     SPS30 device parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                /**< I2C dev the sensor is connected to */
+} sps30_params_t;
+
+/**
+ * @brief     SPS30 device instance
+ */
+typedef struct {
+    sps30_params_t p;             /**< parameters of the sensor device */
+} sps30_t;
+
+/**
+ * @brief     Set of measured particulate matter values
+ *
+ * @warning   Do not change the member order, as it will break the code that
+ *            populates the values in #sps30_read_measurement()
+ *
+ */
+typedef struct {
+    float mc_pm1;       /**< Mass concntration of PM 1.0 [µg/m^3] */
+    float mc_pm2_5;     /**< Mass concntration of PM 2.5 [µg/m^3] */
+    float mc_pm4;       /**< Mass concntration of PM 4.0 [µg/m^3] */
+    float mc_pm10;      /**< Mass concntration of PM 10 [µg/m^3] */
+    float nc_pm0_5;     /**< Number concntration of PM 0.5 [µg/m^3] */
+    float nc_pm1;       /**< Number concntration of PM 1.0 [µg/m^3] */
+    float nc_pm2_5;     /**< Number concntration of PM 2.5 [µg/m^3] */
+    float nc_pm4;       /**< Number concntration of PM 4.0 [µg/m^3] */
+    float nc_pm10;      /**< Number concntration of PM 10 [µg/m^3] */
+    float ps;           /**< Typical particle size [µm] */
+} sps30_data_t;
+
+/**
+ * @brief     SPS30 error codes (returned as negative values)
+ */
+typedef enum {
+    SPS30_OK = 0,     /**< Everything went fine */
+    SPS30_CRC_ERROR,  /**< The CRC check of received data failed */
+    SPS30_I2C_ERROR   /**< Some I2C opration failed */
+} sps30_error_code_t;
+
+/**
+ * @brief     Seconds the fan cleaning process takes in seconds
+ */
+#define SPS30_FAN_CLEAN_S    (10U)
+
+/**
+ * @brief     Length of serial and article code string
+ */
+#define SPS30_SER_ART_LEN    (32U)
+
+/**
+ * @brief     Default fan auto-clean interval in seconds
+ */
+#define SPS30_DEFAULT_ACI_S  (604800U)
```

Comment:
this might need a larger suffix for 8 and 16 bit platforms, i.e., `LU` or even `LLU` not sure. Could test with `arduino-mega2560`

Reply:
Shouldn't make a difference as for just "U" the smallest datatype that fits the value is used. I still added it because UL is indeed more explicit.